pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -660,3 +676,122 @@ func parseResourceGroupHashFromAnnotation(s *fleetv1beta1.ClusterResourceSnapsho
+			Type:               string(fleetv1beta1.ClusterResourcePlacementAppliedConditionType),
+			Reason:             ApplyPendingReason,
+			Message:            "Scheduling has not completed",
+			ObservedGeneration: crp.Generation,
+		}
+		meta.SetStatusCondition(&status.Conditions, appliedCondition)
+		// skip populating detailed resourcePlacementStatus
+		return status, nil
+	}
+
+	// TODO collect work status and build appliedCondition & resourcePlacementStatus
+	status.PlacementStatuses = buildResourcePlacementStatus(crp, latestSchedulingPolicySnapshot)
+	return status, nil
+}
+
+func buildScheduledCondition(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot) metav1.Condition {
+	scheduledCondition := latestSchedulingPolicySnapshot.GetCondition(string(fleetv1beta1.PolicySnapshotScheduled))
+
+	if scheduledCondition == nil ||
+		// defensive check and not needed for now as the policySnapshot should be immutable.
+		scheduledCondition.ObservedGeneration < latestSchedulingPolicySnapshot.Generation ||
+		// We have numberOfCluster annotation added on the CRP and it won't change the CRP generation.
+		// So that we need to compare the CRP observedCRPGeneration reported by the scheduler.
+		latestSchedulingPolicySnapshot.Status.ObservedCRPGeneration < crp.Generation ||
+		scheduledCondition.Status == metav1.ConditionUnknown {
+		return metav1.Condition{
+			Status:             metav1.ConditionUnknown,
+			Type:               string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType),
+			Reason:             "Scheduling",
+			Message:            "Record the intermediate status of the scheduling",
+			ObservedGeneration: crp.Generation,
+		}
+	}
+	return metav1.Condition{
+		Status:             scheduledCondition.Status,
+		Type:               string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType),
+		Reason:             scheduledCondition.Reason,
+		Message:            scheduledCondition.Message,
+		ObservedGeneration: crp.Generation,
+	}
+}
+
+func buildResourcePlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot) []fleetv1beta1.ResourcePlacementStatus {
+	res := make([]fleetv1beta1.ResourcePlacementStatus, 0, len(latestSchedulingPolicySnapshot.Status.ClusterDecisions))
+	for i, c := range latestSchedulingPolicySnapshot.Status.ClusterDecisions {
+		scheduledCondition := metav1.Condition{
+			Status:             metav1.ConditionTrue,
+			Type:               string(fleetv1beta1.PlacementScheduledConditionType),
+			Reason:             "ScheduleSucceeded",
+			Message:            fmt.Sprintf(resourcePlacementConditionScheduleSucceededMessageFormat, c.ClusterName, c.Reason),
+			ObservedGeneration: crp.Generation,
+		}
+		var rp fleetv1beta1.ResourcePlacementStatus
+		if !latestSchedulingPolicySnapshot.Status.ClusterDecisions[i].Selected {
```

Comment:
why is loop on both index and take an loop var? Here the index is used to get the element while  at line 773 the loop var is used.